### PR TITLE
feat(rest-api-client): Add method of space.updateSpace

### PR DESCRIPTION
**Pull request by shabaraba**
## Why

- Support space.updateSpace method to be able to update a space settings.

## What

- Add new method, space.updateSpace
- Add type for UpdateSpace method request
- Add unit test

## How to test

```ts
import {KintoneRestAPIClient} from '@kintone/rest-api-client';

const client = new KintoneRestAPIClient({
  baseUrl: 'http://xxx',
  proxy: false,
  auth: {
    username: 'xxx',
    password: 'xxx',
  },
});

const resp = await client.space.getSpace({id: 1});
console.log(resp);
await client.space.updateSpace({
  id: resp.id,
  name: "updated",
  isPrivate: true,
  useMultiThread: true,
  fixedMember: true,
  showAnnouncement: true,
  showThreadList: true,
  showAppList: true,
  showMemberList: true,
  showRelatedLinkList: true,
  permissions: {
    createApp: "ADMIN",
  },
})

console.log(await client.space.getSpace({id: resp.id}));

```

## Checklist

- [ ] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/main/CONTRIBUTING.md)
- [ ] Updated documentation if it is required.
- [ ] Added tests if it is required.
- [ ] Passed `pnpm lint` and `pnpm test` on the root directory.

### CODE REVIEW

File: packages/rest-api-client/src/client/types/space/index.ts [hunk 1]
```diff
@@ -43,6 +43,20 @@ export type Space = {
...
+  name: string;
+  isPrivate: boolean;
+  useMultiThread: boolean;
+  fixedMember: boolean;
+  showAnnouncement: boolean;
+  showThreadList: boolean;
+  showAppList: boolean;
+  showMemberList: boolean;
+  showRelatedLinkList: boolean;
+  permissions: Permissions;
```
Comment (tasshi-me) [inline packages/rest-api-client/src/client/types/space/index.ts:57]: スペースID以外は省略可能なのでOptionalですね！

```suggestion
  name?: string;
  isPrivate?: boolean;
  useMultiThread?: boolean;
  fixedMember?: boolean;
  showAnnouncement?: boolean;
  showThreadList?: boolean;
  showAppList?: boolean;
  showMemberList?: boolean;
  showRelatedLinkList?: boolean;
  permissions?: Permissions;
```

https://cybozu.dev/ja/kintone/docs/rest-api/spaces/update-space/#request

File: packages/rest-api-client/docs/space.md [hunk 2]
```diff
@@ -91,6 +92,35 @@ Gets general information of a space.
...
+| showAppList           |     Boolean      |          | The display status for the Apps widget.<br /><strong>true</strong>: The Apps widget is displayed.<br /><strong>false</strong>: The Apps widget is not displayed.<br /><strong>null</strong> is returned for Spaces with the Enable multiple threads option turned off.                                                    |
+| showMemberList        |     Boolean      |          | The display status for the People widget.<br /><strong>true</strong>: The People widget is displayed.<br /><strong>false</strong>: The People widget is not displayed.<br /><strong>null</strong> is returned for Spaces with the Enable multiple threads option turned off.                                              |
+| showRelatedLinkList   |     Boolean      |          | The display status for the Related Apps & Spaces widget.<br /><strong>true</strong>: The Related Apps & Spaces widget is displayed.<br /><strong>false</strong>: The Related Apps & Spaces widget is not displayed.<br /><strong>null</strong> is returned for Spaces with the Enable multiple threads option turned off. |
```
Comment (tasshi-me) [inline packages/rest-api-client/docs/space.md:112]: こちらはUpdate側なので`null`の説明は不要そうです！

**Review by tasshi-me — CHANGES_REQUESTED** [review]
2箇所レビュー追加しました！ 👍

File: packages/rest-api-client/src/client/types/space/index.ts (hunk 1 above)
Comment (shabaraba) [reply packages/rest-api-client/src/client/types/space/index.ts]: ありがとうございます、見逃してました
[89e6ee6](https://github.com/kintone/js-sdk/pull/2884/commits/89e6ee60c50ae58831bfe26484917a7cb28ecc69)

File: packages/rest-api-client/docs/space.md (hunk 2 above)
Comment (shabaraba) [reply packages/rest-api-client/docs/space.md]: [89e6ee6](https://github.com/kintone/js-sdk/pull/2884/commits/89e6ee60c50ae58831bfe26484917a7cb28ecc69)

**Review by tasshi-me — APPROVED** [review]
LGTM!

**Pull request merged**